cpu/lm4f120/periph/uart.c

```diff
@@ -31,120 +31,217 @@
  */
 static uart_isr_ctx_t config[UART_NUMOF];
 
-static int init_base(uart_t uart, uint32_t baudrate);
+typedef struct {
+    unsigned long uart_sysctl;         /**< UART device in sysctl */
+    unsigned long uart_base;           /**< UART base address */
+    IRQn_Type uart_irq_chan;           /**< UART IRQ channel */
+    volatile unsigned long * uart_im_r;/**< UART RX Enable Register */
+
+    unsigned long gpio_sysctl;         /**< GPIO device in sysctl */
+    unsigned long gpio_port;           /**< GPIO port */
+    unsigned long rx;              /**< pin used for RX */
+    unsigned long tx;              /**< pin used for TX */
+    unsigned long mask_rx;            /**< Pin mask for RX*/
+    unsigned long mask_tx;            /**< Pin mask for TX*/
+} uart_conf_fixed_t;
+
+static const uart_conf_fixed_t ucf[] = {
+    {
+        .uart_sysctl    = SYSCTL_PERIPH_UART0,
+        .uart_base      = UART0_BASE,
+        .uart_irq_chan  = UART0_IRQn,
+        .uart_im_r      = &UART0_IM_R,
+
+        .gpio_sysctl = SYSCTL_PERIPH_GPIOA,
```

Comment:
Have you checked if it would make sense to factor the pin confguration out and move it to the periph/gpio driver (a common practice used for many other CPUs)?

That way, you could simply add two fields for RX and TX pins to the config struct in the board configuration, e.g. 
```c 
   ... // UART_DEV(0) config
    .pin_rx = GPIO_PIN(PA, 0),
    .pin_tx = GPIO_PIN(PA, 1),
    ...
```
and then call something like `gpio_init_sysctl(uart_config[dev].pin_rx, SYSCTL_UARTx)` or similar from the UART init function...